### PR TITLE
#317: Radio-group component

### PR DESCRIPTION
**Pull request by BeeMargarida**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-design/issues/317 |
| Dependencies | https://github.com/ripe-tech/ripe-components-react-native/pull/179 |
| Decisions | Radio Group component |
| Animated GIF | radio-group-iosradio-group-android|

### CODE REVIEW

File: react/components/molecules/radio-group/radio-group.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    static get defaultProps() {
+        return {
+            items: [],
+            values: {},
+            error: false,
+            disabled: false,
+            beforeItem: undefined,
+            afterItem: undefined,
+            onUpdateValue: () => {},
```
Comment (beneditomauro) [inline react/components/molecules/radio-group/radio-group.js:39]: default with parameter

File: react/components/molecules/radio-group/radio-group.stories.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
+import React, { PureComponent } from "react";
+import { Text, View } from "react-native";
+import { storiesOf } from "@storybook/react-native";
+import { withKnobs, boolean, select } from "@storybook/addon-knobs";
+import PropTypes from "prop-types";
```
Comment (beneditomauro) [inline react/components/molecules/radio-group/radio-group.stories.js:5]:
```suggestion
import PropTypes from "prop-types";
import { storiesOf } from "@storybook/react-native";
import { withKnobs, boolean, select } from "@storybook/addon-knobs";

```

File: react/components/molecules/radio-group/radio-group.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                        item: item,
+                        checked: item.value === this.state.valueData
+                    })}
+            </View>
+        ));
+    };
+
+    render() {
+        return (
+            <View style={[styles.radioGroup, this.props.style]} {...this.id("radio-group")}>
```
Comment (beneditomauro) [inline react/components/molecules/radio-group/radio-group.js:97]: Even tho there's no conditionals in setting style, maybe move all style to a `_style()` const still

File: react/components/molecules/radio-group/radio-group.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                        item: item,
+                        checked: item.value === this.state.valueData
+                    })}
+            </View>
+        ));
+    };
+
+    render() {
+        return (
+            <View style={[this._style()]} {...this.id("radio-group")}>
```
Comment (beneditomauro) [inline react/components/molecules/radio-group/radio-group.js:101]:
```suggestion
            <View style={this._style()} {...this.id("radio-group")}>

```

**Review by beneditomauro — APPROVED** [review]

File: react/components/molecules/radio-group/radio-group.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            valueData: props.value
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.value === this.props.value) return;
```
Comment (gcandal) [inline react/components/molecules/radio-group/radio-group.js:53]: Brackets and not one-liner.

File: react/components/molecules/radio-group/radio-group.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+    _style() {
+        return [styles.radioGroup, this.props.style];
+    }
+
+    _renderRadios = () => {
+        return this.props.items.map((item, index) => (
+            <View style={styles.radioItem} key={item.value}>
+                {this.props.beforeItem &&
+                    React.cloneElement(this.props.beforeItem, {
```
[the remark this answers is not in the document]
Comment (BeeMargarida) [reply react/components/molecules/radio-group/radio-group.js:76]: Its the only way to pass arguments to slots. If no info needed to be passed we could only do `{this.props.beforeItem}`.

[the remark this answers is not in the document]
Comment (gcandal) [reply react/components/molecules/radio-group/radio-group.js:76]: Ok got it, I remember it now (also: https://github.com/ripe-tech/ripe-components-react-native/blob/4929a27fb8a84d4c7e132899fc1d0066842973a1/react/components/atoms/container-draggable/container-draggable.js#L71)

**Review by gcandal — APPROVED** [review]

**Review by joamag — APPROVED** [review]

**Pull request merged**